geode-core/src/main/java/org/apache/geode/internal/InternalDataSerializer.java

```diff
@@ -2094,8 +2030,8 @@ public static void basicWriteObject(Object o, DataOutput out, boolean ensurePdxC
         out.writeByte(DSCODE.DATA_SERIALIZABLE.toByte());
         DataSerializer.writeClass(c, out);
       }
-      DataSerializable ds = (DataSerializable) o;
-      invokeToData(ds, out);
+      final DataSerializable bs = (DataSerializable) o;
```

Comment:
why rename this to "bs" since it is still a DataSerializable?

Reply:
Actually we don't need that cast at all since `invokeToData()` takes an `Object`. Fixing…